### ISSUE TITLE
Add APIs to check if an actor is in contact with another actor or block

### ISSUE THREAD

**Issue by JohnSundell** [JohnSundell/ImagineEngine#109]
It would be really nice to be able to check if an actor is currently in contact with another actor or block, like this:

```swift
let isInContactWIthOtherActor = actor.isInContact(with: anotherActor)
let isInContactWithBlock = actor.isInContact(with: block)
```

Thankfully this is super easy to implement, since `Actor` already has internal properties that keep track of its current contacts:

```swift
internal lazy var actorsInContact = Set<Actor>()
internal lazy var blocksInContact = Set<Block>()
```

**Things to do**

- Implement the two methods that enables the API user to check if an actor is currently in contact with another actor or block.
- Use the contact sets to return true or false.
- Unit test by moving an actor into contact with another actor and block and verify that the method returns the correct result.

**Comment by z0s** [comment]
I'd like to take this issue!

**Comment by JohnSundell** [comment]
I think it's fair to say that this one is again up for grabs if anyone wants to take it 🙂

**Comment by JohnSundell** [comment]
Implemented by @pablocarmu in #155 🎉
